### PR TITLE
porting pblintd_init function

### PR DESCRIPTION
**Pull request by xyuan**
porting pblintd_init function

[BFB]

### CODE REVIEW

**Review by jeff-cohere — APPROVED** [review]
Looks good.

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp [hunk 1]
```diff
@@ -2417,6 +2427,57 @@ void pblintd_height_f(Int shcol, Int nlev, Real* z, Real* u, Real* v, Real* usta
...
+    Scalar& z_s           = z_1d(nlev_pack)[nlev_indx];
+
+    SHOC::pblintd_init(z_s, check_s, rino_s, pblh_s);
+
+    check_d(i)   = check_s;
+    pblh_d(i)[0] = pblh_s;
+  });
+
+  Kokkos::Array<view_2d, 1> out_2d_views = {rino_2d};
+  ekat::device_to_host<int, 1>({rino}, {shcol}, {nlev}, out_2d_views, true);
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/shoc_functions_f90.cpp:2470]: You don't need to use the API that takes vectors of sizes if they are all the same.

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp [hunk 2]
```diff
@@ -2417,6 +2427,57 @@ void pblintd_height_f(Int shcol, Int nlev, Real* z, Real* u, Real* v, Real* usta
...
+
+  Kokkos::Array<view_1d, 1> out_1d_views = {pblh_d};
+  ekat::device_to_host<int, 1>({pblh}, {shcol}, out_1d_views);
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/shoc_functions_f90.cpp:2473]: Same as above.

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp [hunk 3]
```diff
@@ -2417,6 +2427,57 @@ void pblintd_height_f(Int shcol, Int nlev, Real* z, Real* u, Real* v, Real* usta
...
+
+  Kokkos::deep_copy(host_check_d, check_d);
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/shoc_functions_f90.cpp:2475]: device_to_host now works for bools so you should use that instead. In fact, I think it has always works for 1d data.

**Review by jgfouca — CHANGES_REQUESTED** [review]
Minor changes needed to _f.

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp (hunk 3 above)
Comment (jgfouca) [reply components/scream/src/physics/shoc/shoc_functions_f90.cpp:2917]: I was wrong, device_to_host expects packed data.

**Review by jgfouca — APPROVED** [review]

**Pull request merged**